internal/backend/remote-state/azure/client.go

```diff
@@ -84,10 +84,8 @@ func (c *RemoteClient) Put(data []byte) error {
 	}
 
 	blob, err := c.giovanniBlobClient.GetProperties(ctx, c.accountName, c.containerName, c.keyName, getOptions)
-	if err != nil {
-		if blob.StatusCode != 404 {
-			return err
-		}
+	if err != nil && blob.Response.Response != nil && blob.Response.Response.StatusCode != 404 {
```

Comment:
We should probably look to use the `response.WasNotFound(blob.Response.Response)` from https://github.com/hashicorp/go-azure-helpers here rather than the multiple if conditions? e.g.
```
	blob, err := c.giovanniBlobClient.GetProperties(ctx, c.accountName, c.containerName, c.keyName, getOptions)
	if err != nil {
		if !response.WasNotFound(blob.Response.Response) {
			return err
		}
	}
```